Data-Structures/Linked-List/CycleDetectionII.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * A LinkedList based solution for Detecting a start node of the Cycle in a list.
+ * https://en.wikipedia.org/wiki/Cycle_detection
+ */
+
+function detectCycleNode(head) {
+  /*
+  Problem Statement:
+  Given the head of a linked list, return the node where the cycle begins. If there is no cycle, return null.
+  Link for the Problem: https://leetcode.com/problems/linked-list-cycle-ii/
+  */
+
+  let length = 0
+  let fast = head
+  let slow = head
+
+  while (fast !== null && fast.next !== null) {
+    fast = fast.next.next
+    slow = slow.next
+    if (fast === slow) {
+      // Calculate length of the cycle
+      length = lengthCycle(slow)
+      break
+    }
+  }
+
+  if (length === 0) {
+    // If length = 0, return null
+    return null
+  }
+
+  //Point the nodes at head
```

Comment:
Redundant comment. (Also, could be shortened to `fast = slow = head` in principle.)

Reply:
Yes. `fast = slow = head` would be better. I have made this change as well.